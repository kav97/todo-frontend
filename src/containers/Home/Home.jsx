import "./Home.scss";
import { faCirclePlus } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import ToDoCardContainer from "../ToDoCardContainer/ToDoCardContainer";

const Home = () => {
  return (
    <div className="home-page">
      <button
        className="home-page__create-button"
        aria-label="Navigate to create a to-do"
        onClick={() => {
          window.location.href = `todo-frontend/create`;
        }}
      >
        <FontAwesomeIcon icon={faCirclePlus} />
      </button>
      <ToDoCardContainer />
    </div>
  );
};

export default Home;