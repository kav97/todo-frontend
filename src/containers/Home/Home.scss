@use "../../assets/styles/variables.scss" as *;

.home-page {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__create-button {
    display: flex;
    flex-direction: column;
    height: 2.2rem;
    width: 2.2rem;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    background-color: $color-white;
    color: $color-black;
    border: none;
    cursor: pointer;

    &:hover {
      background-color: $color-black;
      color: $color-white;
    }
  }
}