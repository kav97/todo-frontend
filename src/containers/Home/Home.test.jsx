import { render } from "@testing-library/react";
import Home from "./Home";

it("Should render the home page to screen", () => {
    const { container } = render(<Home />);
  
    expect(container).toMatchSnapshot();
});