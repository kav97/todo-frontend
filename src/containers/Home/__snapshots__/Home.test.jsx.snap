// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the home page to screen 1`] = `
<div>
  <div
    class="home-page"
  >
    <button
      aria-label="Navigate to create a to-do"
      class="home-page__create-button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-plus "
        data-icon="circle-plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM232 344V280H168c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V168c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H280v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="empty-list"
    >
      <p
        class="empty-list__message"
      >
        Oh nooo! There appears to be nothing to do today. Why don't you add something to your list? ☝️
      </p>
      <img
        alt="Yikes nothing to do"
        class="empty-list__img"
        src="no-work.gif"
      />
    </div>
  </div>
</div>
`;
